x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -153,16 +153,21 @@ export const createLifecycleExecutor = (
 
   const commonRuleFields = getCommonAlertFields(options);
 
-  const currentAlerts: Record<string, ExplicitAlertFields> = {};
+  const currentAlerts: Record<
+    string,
+    { indexName: string; fields: Partial<ExplicitAlertFields> }
+  > = {};
 
   const lifecycleAlertServices: LifecycleAlertServices<
     InstanceState,
     InstanceContext,
     ActionGroupIds
   > = {
     alertWithLifecycle: ({ id, fields }) => {
-      currentAlerts[id] = fields;
-
+      currentAlerts[id] = {
+        indexName: ruleDataClient.indexName,
```

Comment:
@weltenwort I get indexName from ruleDataClient here. I had to do so to fix an `Error: Cannot read property 'fields' of undefined` which makes sense since I changed the data structure to populate indexName and fields properties.

How about I undo these changes and get indexName directly from ruleDataClient in the bulk operation?